### DISCUSSION TITLE
5 character hex code

### DISCUSSION THREAD

**Discussion by red-tails** [Tyrrrz/DiscordChatExporter#583] [General]
Filenames look like this:

DSC_1140-B2B32.JPG
20210310_174625-9CB58.jpg
1407354073407-1380C.gif
dvgrab-01_01_07_18-452C5.webm

Why is there a 5 character hex code after each file name? Is this to prevent duplicate filenames? If so, what is this hex code based on?

**Comment by 96-LB** [comment]
> Is this to prevent duplicate filenames?

Yup. Discord channels can have a lot of files with the same name, such as "unknown" or "image0".

> If so, what is this hex code based on?

Quoting the changelog, "a URL hash is injected in every file name". This makes it so that the same file always has the same filename when downloaded so that if you enable the "reuse media" option, DiscordChatExporter knows which files to ignore.

**Reply by red-tails** [reply]
I am not understanding what "reuse media" does. When I enabled it, it would not prevent hash-duplicates nor reuse media between DMs and/or servers

**Reply by 96-LB** [reply]
If you enable "reuse media" and export the same channel twice (for example, you might re-export a channel after a week to update your backup to include the latest messages), the program won't download any files you already have stored. It doesn't work across channels, however, because each export contains its media in a different folder.
